### ISSUE TITLE
Problems with the prediciton

### ISSUE THREAD

**Issue by qiuwei** [nlpub/hyperstar#2]
I can not fully understand the setup of the experiment.

The clustering criteria is the |Vector_hyponym - Vector_hypernym|,
During test time, only the hyponym is known. Then how can you decide which cluster the relation should be in and which projection matrix should be used?

**Comment by dustalov** [comment]
This problem with clustering comes from the baseline described in [Fu et al.](http://aclweb.org/anthology/P/P14/P14-1113.pdf) (2014).

In our experiments, we have train, validation, and test sets of subsumption pairs. Given the test pair (*x*, *y*), we first compute the offset y ⃗−x ⃗ and retrieve the corresponding cluster *i*. Note that the clusters are learned only on the training data. Then, we apply the transformation matrix corresponding to the *i*-th cluster to x ⃗ and extract the *k* nearest neighbors of the transformed vector. According to the hit@k measure, we treat the pair as correctly evaluated if the set of those neighbors contains *y*.

**Comment by qiuwei** [comment]
I understand the process. 

However, I still consider the problem oversimplified. The whole point is to predict the hypernyms, but your algorithm depends on the fact that the hypernym(or at least one of them) is already known.

Thanks anyway.

**Comment by dustalov** [comment]
During the prediction after the cluster estimation, we have no information about the hypernym we have. In fact, this evaluation approach differs from the “classical” binary classification. Of course, the hit@k measure is less strict than, for instance, accuracy or precision and recall.
